src/generate-meta.ts

```diff
@@ -0,0 +1,295 @@
+import {
+  basicTypeName,
+  detectMapType,
+  isMessage,
+  isRepeated,
+  isWithinOneOf,
+  messageToTypeName,
+  responseType,
+  toReaderCall,
+  TypeMap,
+} from './types';
+import SourceInfo from './sourceInfo';
+import {
+  Any,
+  CodeBlock,
+  InterfaceSpec,
+  Member,
+  Modifier,
+  PropertySpec,
+  TypeAliasSpec,
+  TypeName,
+  TypeNames,
+  TypeVariable,
+  Union,
+} from 'ts-poet';
+import { maybeSnakeToCamel } from './case';
+import { OneofOption, Options } from './main';
+import { google } from '../build/pbjs';
+import DescriptorProto = google.protobuf.DescriptorProto;
+import FileDescriptorProto = google.protobuf.FileDescriptorProto;
+import ServiceDescriptorProto = google.protobuf.ServiceDescriptorProto;
+import FieldDescriptorProto = google.protobuf.FieldDescriptorProto;
+
+export function getMetaInterfaces() {
```

Comment:
This is a good approach for this PR, and maybe even indefinitely. Wdyt about maybe eventually having like a separate `ts-proto-schemas` or some sort of separate package that we could put these types into and import?

As a disclaimer, I know I've shied away from runtime dependencies, i.e. with things like `DeepPartial`, but those seemed smaller, and were also used by all clients of ts-proto, and not just ones who are opting into something specific/special like your runtime metadata/schema.

And yeah, let's keep this for this PR, just thinking of maybe later.